File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/owner/OwnerDTO.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.owner;

import jakarta.validation.constraints.NotNull;
import java.time.OffsetDateTime;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class OwnerDTO {

    private Integer id;

    @NotNull
    private String firstName;

    @NotNull
    private String lastName;

    @NotNull
    private String address;

    @NotNull
    private String city;

    @NotNull
    private String telephone;

    @NotNull
    private String email;

    @NotNull
    private OffsetDateTime createdAt;

    @NotNull
    private OffsetDateTime updatedAt;

    private OffsetDateTime deletedAt;

}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/owner/OwnerDTO.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.owner

import jakarta.validation.constraints.NotNull
import org.springframework.format.annotation.DateTimeFormat
import java.time.OffsetDateTime

class OwnerDTO {

    var id: Int? = null

    @NotNull
    var firstName: String? = null

    @NotNull
    var lastName: String? = null

    @NotNull
    var address: String? = null

    @NotNull
    var city: String? = null

    @NotNull
    var telephone: String? = null

    @NotNull
    var email: String? = null

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm:ssXXX")
    var createdAt: OffsetDateTime? = null

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm:ssXXX")
    var updatedAt: OffsetDateTime? = null

    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm:ssXXX")
    var deletedAt: OffsetDateTime? = null
}
